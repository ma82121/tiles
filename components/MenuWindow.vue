<template>
  <div v-show="isShow" class="menu-window">
    <div class="menu-window__inner">
      <div class="menu-window__title">
        {{ menuTitle }}
      </div>
      <TabList :list="menuList" :current-id="selectedId" @clickTab="clickTab" />
      <MenuWindowConfig v-show="selectedId === 'config'" />
      <MenuWindowOnline v-show="selectedId === 'online'" />
      <MenuWindowSelectStage v-show="selectedId === 'select-stage'" />
      <MenuWindowTutorial v-show="selectedId === 'tutorial'" />
    </div>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'

import TabList from '@/components/TabList'
import MenuWindowConfig from '@/components/MenuWindowConfig'
import MenuWindowOnline from '@/components/MenuWindowOnline'
import MenuWindowSelectStage from '@/components/MenuWindowSelectStage'
import MenuWindowTutorial from '@/components/MenuWindowTutorial'

import { MENU_LIST } from '@/assets/scripts/dictionary/MENU_LIST'

export default {
  name: 'MenuWindow',
  components: {
    TabList,
    MenuWindowConfig,
    MenuWindowOnline,
    MenuWindowSelectStage,
    MenuWindowTutorial
  },
  computed: {
    ...mapGetters('menu', ['isShow', 'selectedId']),
    menuTitle() {
      return MENU_LIST.find(item => item.id === this.selectedId).title
    },
    menuList() {
      return MENU_LIST
    }
  },
  methods: {
    ...mapActions('menu', ['setMenuSelectedId']),
    clickTab(string) {
      this.setMenuSelectedId(string)
    }
  }
}
</script>

<style lang="scss" scoped>
.menu-window {
  position: fixed;
  padding: 10px;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  background-color: #fff;
  z-index: 80;
  overflow: auto;
}

.menu-window__inner {
  margin: 0 auto;
  padding-bottom: 40px;
  max-width: 800px;
  font-size: 0.9rem;
  text-align: left;
}

.menu-window__title {
  margin-bottom: 10px;
  height: 60px;
  line-height: 60px;
  font-size: 1.6rem;
  text-align: center;
  font-weight: bold;
  color: #1cc4fd;
}
</style>
