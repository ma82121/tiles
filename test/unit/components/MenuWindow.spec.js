import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import firebase from '@/plugins/firebase'
import MenuWindow from '@/components/MenuWindow'

import {
  state as configState,
  mutations as configMutations,
  getters as configGetters,
  actions as configActions
} from '@/store/config'

import {
  state as menuState,
  mutations as menuMutations,
  getters as menuGetters,
  actions as menuActions
} from '@/store/menu'

import {
  state as singlePlayState,
  mutations as singlePlayMutations,
  getters as singlePlayGetters,
  actions as singlePlayActions
} from '@/store/singlePlay'

import {
  state as authState,
  mutations as authMutations,
  getters as authGetters,
  actions as authActions
} from '@/store/auth'

const localVue = createLocalVue()
localVue.use(Vuex, VueRouter)

describe('MenuWindow', () => {
  let store
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        config: {
          namespaced: true,
          state: configState,
          mutations: configMutations,
          getters: configGetters,
          actions: configActions
        },
        menu: {
          namespaced: true,
          state: menuState,
          mutations: menuMutations,
          getters: menuGetters,
          actions: menuActions
        },
        singlePlay: {
          namespaced: true,
          state: singlePlayState,
          mutations: singlePlayMutations,
          getters: singlePlayGetters,
          actions: singlePlayActions
        },
        auth: {
          namespaced: true,
          state: authState,
          mutations: authMutations,
          getters: authGetters,
          actions: authActions
        }
      }
    })
  })
  test('snapshot', () => {
    const wrapper = shallowMount(MenuWindow, {
      store,
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
