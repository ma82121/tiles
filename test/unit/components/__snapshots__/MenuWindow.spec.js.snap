// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuWindow snapshot 1`] = `
<div
  class="menu-window"
  style="display: none;"
>
  <div
    class="menu-window-inner"
  >
    <div
      class="menu-title"
    >
      
      遊び方
    
    </div>
     
    <tablist-stub
      currentid="tutorial"
      list="[object Object],[object Object],[object Object],[object Object]"
    />
     
    <div
      class="menu-window-content"
    >
      <menuwindowconfig-stub
        style="display: none;"
      />
       
      <menuwindowonline-stub
        style="display: none;"
      />
       
      <menuwindowselectstage-stub
        style="display: none;"
      />
       
      <menuwindowtutorial-stub />
    </div>
  </div>
</div>
`;
